pkg/resource/validations/kmsArnRegexValidation.go

```diff
@@ -0,0 +1,19 @@
+package validations
+
+import (
+	"fmt"
+	"regexp"
+)
+
+var KmsArnRE = regexp.MustCompile(
+	`^arn:aws[\w-]*:kms:[\w-]+:\d{12}:key\/mrk-[0-9a-f]{32}$|[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$`,
+)
+
+func ValidateKMSKeyARN(kmsKeyARN *string) error {
+	if kmsKeyARN != nil {
+		if !KmsArnRE.MatchString(*kmsKeyARN) {
+			return fmt.Errorf("expected a valid value for kms-key-arn matching %s", KmsArnRE)
```

Comment:
@den-rgb I don't know how useful this error message is to the user :-) Maybe it's just better to do something like:

```
return fmt.Errorf("The KMS Key ARN '%s' is not a valid ARN", kmsKeyARN)
```

I personally think that's much clearer for the user where the problem is rather than them trying to grok the regex and figure out where the problem might be.

WDYT?

Reply:
the point of the code is to check if it matches the regexp, so I still think it would be beneficial to include an example of some sort of why it's wrong, so I could maybe add an example arn?